### PR TITLE
Look for song change when we tick for lyrics

### PR DESCRIPTION
**Pull request by varun7654**
Hopefully resolves issues with the old method.
I've also removed the delay when adding lyrics, which should make loading songs much faster. It now pass the song artist to the funcs that need them to be able to remove some extra code.

Change: Don't reload lyrics when clicking the lyrics tab, when lyrics are already loaded.
Also fixed scrolling issues when initially loading the lyrics.

## Type of change

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Checklist

- [x] I have performed a self-review of my code

### CODE REVIEW

**Comment by varun7654** [comment]
Also made a few changes to how the loader is displayed & when we clear lyrics to resolve some flickering issues

File: src/index.js [hunk 1]
```diff
@@ -21,21 +24,76 @@ BetterLyrics.App = {
...
   init: function () {
     try {
       if (document.readyState !== "loading") {
         BetterLyrics.App.modify();
         document.addEventListener("blyrics-send-player-time", function (event) {
-          BetterLyrics.DOM.tickLyrics(event.detail.currentTime);
+          let detail = event.detail;
+
+          // Video ID can change before the song & artist are available; also encode them, so we listen to when they change
+          // If the song and artist are available it seems to be safe to inject our lyrics w/o any sort of delay
```
Comment (adaliea) [inline src/index.js]: outdated

File: src/index.js [hunk 2]
```diff
@@ -21,21 +24,76 @@ BetterLyrics.App = {
...
+          let currentVideoId = detail.videoId;
+          let currentVideoDetails = detail.song + " " + detail.artist;
+
+          if (currentVideoId !== BetterLyrics.App.lastVideoId || currentVideoDetails !== BetterLyrics.App.lastVideoDetails) {
+            try {
+              if (currentVideoId === BetterLyrics.App.lastVideoId && BetterLyrics.App.areLyricsTicking) {
+                console.log("[BetterLyrics] Skipping Reload From Detail Load: Already Loaded");
```
Comment (adaliea) [inline src/index.js]: move to constant

File: src/index.js [hunk 3]
```diff
@@ -21,21 +24,76 @@ BetterLyrics.App = {
...
+                return; // We already loaded this video
+              }
+            } finally {
+              BetterLyrics.App.lastVideoId = currentVideoId;
+              BetterLyrics.App.lastVideoDetails = currentVideoDetails;
+            }
+
+            if (!detail.song || !detail.artist) {
+              console.log("[BetterLyrics] Trying to load without Song/Artist info");
```
Comment (adaliea) [inline src/index.js]: move to constants

File: src/index.js [hunk 4]
```diff
@@ -21,21 +24,76 @@ BetterLyrics.App = {
...
+              BetterLyrics.DOM.addAlbumArtToLayout,
+              BetterLyrics.DOM.removeAlbumArtFromLayout
+            );
+
+            const tabSelector = document.getElementsByClassName(BetterLyrics.Constants.TAB_HEADER_CLASS)[1];
+            if (!tabSelector || tabSelector.getAttribute("aria-selected") === "true") {
+              BetterLyrics.Utils.log(BetterLyrics.Constants.LYRICS_TAB_VISIBLE_LOG);
+              BetterLyrics.App.handleModifications(detail.song, detail.artist, detail.currentTime, detail.videoId);
+            } else {
+              BetterLyrics.Settings.onAutoSwitchEnabled(() => {
```
Comment (adaliea) [inline src/index.js]: This'll cause issues if auto switch is enabled

File: src/modules/ui/observer.js [hunk 5]
```diff
@@ -26,63 +26,18 @@ BetterLyrics.Observer = {
...
 
     if (tab1 !== undefined && tab2 !== undefined && tab3 !== undefined) {
       tab2.addEventListener("click", function () {
-        BetterLyrics.Utils.log(BetterLyrics.Constants.LYRICS_TAB_CLICKED_LOG);
-        BetterLyrics.App.handleModifications();
+        if (!BetterLyrics.App.areLyricsLoaded) {
+          BetterLyrics.Utils.log(BetterLyrics.Constants.LYRICS_TAB_CLICKED_LOG);
+          BetterLyrics.DOM.cleanup();
+          BetterLyrics.App.reloadLyrics();
+        }
```
Comment (adaliea) [inline src/modules/ui/observer.js]: Usually this will mean we don't have a cached version; display the loader here aswell?

**Review by varun7654 — COMMENTED** [review]
notes on stuff I need to change

**Comment by boidushya** [comment]
2 things @varun7654 
- The loader animation is no longer working
- Seems like auto switch to lyrics tab is broken?

**Comment by boidushya** [comment]
Moved to #92

**Pull request closed without merging**